test/valid-data/type-intersection-union-recursive-interface/main.ts

```diff
@@ -0,0 +1,15 @@
+export interface Container {
+    child: Union;
+}
+
+export interface Dummy {
+    x: number;
+}
+
+export interface Silly {
+    y: number;
+}
+
+export type Union = Container | Silly;
```

Comment:
Let's just test two things. First, what if we inline this type or not export it. Second, what if we define another type that just aliases Union? 

You can test it locally. No need to write a test case unless you notice that it fails.

Reply:
Sure. I tested various combinations of inlining, not exporting, aliasing, etc. No missing definitions occurred and the produced schemas looked okay to me.

Reply:
Hi @domoritz! What do you think, is there still something else to test or adjust?

Reply:
No. I need to do a review. Thank you for your patience while I come back from the break and need to catch up on some other things. 

I bumped up this review in my queue.